x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackClientPlugin.java

```diff
@@ -452,7 +452,11 @@ static Settings additionalSettings(final Settings settings, final boolean enable
                 new NamedXContentRegistry.Entry(LifecycleAction.class, new ParseField(ReadOnlyAction.NAME), ReadOnlyAction::parse),
                 new NamedXContentRegistry.Entry(LifecycleAction.class, new ParseField(RolloverAction.NAME), RolloverAction::parse),
                 new NamedXContentRegistry.Entry(LifecycleAction.class, new ParseField(ShrinkAction.NAME), ShrinkAction::parse),
-                new NamedXContentRegistry.Entry(LifecycleAction.class, new ParseField(DeleteAction.NAME), DeleteAction::parse)
+                new NamedXContentRegistry.Entry(LifecycleAction.class, new ParseField(DeleteAction.NAME), DeleteAction::parse),
+                // ILM - Client - Lifecycle Actions
+                new NamedXContentRegistry.Entry(org.elasticsearch.protocol.xpack.indexlifecycle.AllocateAction.class,
+                    new ParseField(org.elasticsearch.protocol.xpack.indexlifecycle.AllocateAction.NAME),
+                        org.elasticsearch.protocol.xpack.indexlifecycle.AllocateAction::parse)
```

Comment:
I don't think this is going to work, we need to register the `categoryClass` as something more generic than the specific action because when we are deserialising the `Phase` we don't know what specific action we are deserialising (hence why we use the NamedXContentRegistry. I think we need an interface equivalent to `LifecycleAction` but on the client side. 

Also, we will want to move the registering of the server action classes to `IndexLifecycle` since the client won't need access to the server side objects (this can be done in a clean up PR after the client objects are there if you want but its important we don't forget to do it).